Switch to single Channel Model

Harwayne opened the pull request:
## Proposed Changes

- Initial Controllers for the Broker and Trigger.

**Release Note**

<!-- Enter your extended release note in the below block. If the PR requires
additional action from users switching to the new release, include the string
"action required". If no release note is required, write "NONE". -->

```release-note
You need to create the ServiceAccount yourself (`sa.yaml`).
```

Harwayne committed:
Add the Istio injection annotation.

Harwayne committed:
Resolve subscriber in the Trigger controller.

Harwayne committed:
Standardize on 'Any'.

Harwayne committed:
Make Broker and Trigger generational

Harwayne committed:
Happy control path.

Harwayne committed:
TargetPort is 8080

Harwayne committed:
Watch channels.

Harwayne committed:
Custom service account for filter (needs trigger watch).

Harwayne committed:
Increase logging in the Filter to debug level.

n3wscott commented on cmd/broker/ingress/main.go:
```diff
@@ -0,0 +1,147 @@
+	s.Shutdown(ctx)
+}
+
+func getRequiredEnv(envKey string) string {
```
Maybe this will be interesting to you: https://github.com/Botless/slack/blob/master/cmd/rtm/main.go#L13

from: github.com/kelseyhightower/envconfig

Harwayne committed:
Use the default channel provisioner.

nachocano committed:
Adding filtering using k8s label selectors

Harwayne committed:
Watch namespaces and create a default Broker.

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into broker-new-model

nachocano committed:
Updating trigger example with filters

adrcunha committed:
Update test-infra to fix release script (#776)

Harwayne committed:
Broker changes cause the namespace watcher to reconcile.

nachocano committed:
Adding RateLimiter to GCP PubSub channel to backoff retries on dispatch failures  (#775)

* Adding an exponential backoff rate limiter for retrying nack-ed messages.

* Desugar-ing logs. Updating comments.

* Updating autogen code

* Updates after review comments. Adding UTs for retry mechanism.

* Starting loop from 0

* Adding comment back. Loop should start from 1.

matzew committed:
:eyes: make config snippet more readable (#774)

nbarthwal committed:
Update test-infra (#780)

mattmoor-sockpuppet committed:
golang format tools (#783)

Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -type f -name '*.go' -print))`
  `goimports -w $(find -name '*.go' | grep -v vendor)`

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into broker-new-model

Harwayne committed:
Merge branch 'master' into broker-new-model

Harwayne committed:
Move the Logging package import and format.

n3wscott committed:
Refactor controller paths (#786)

* Rename controller to reconciler

For consistency with serving.

* Rename eventing to v1alpha1

For consistency with serving.

* Rename reconcile.go to subscription.go

For consistency with Serving.

* Remove unused file

This is a relic of the copy from serving.

* Move reconciler/names.go to its own package

Avoids naming conflict between reconciler structs and reconciler
package.

* Move inmemory channel to pkg/provisioners

* Move provider to same file as reconciler

For consistency with Serving.

* Update package name in contrib channels

Code in reconciler package moved to reconciler/names.

* Update location of inmemory channel controller

Moved to pkg/provisioners.

* Update dependencies and codegen

* Adding a simple test for controller.

* adding another bad test.

* ran hack/update-codegen.sh

* adding time to the channel reconcile timeout.

* Fix links.

nachocano committed:
Updating after review comments. Only doing exact header matching.
In a follow up PR, I plan to include matching of expressions. It turns
our that they are somewhat more involved than expected.

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into broker-new-model

nachocano committed:
Adding filtering expressions. Currently using LabelSelectors without
validations. Using reflection to set some unexposed fields for now.

vagababov committed:
change the use of map[string]bool to use k8s string sets. (#789)

* change the use of map[string]bool to use k8s string sets.

This is part of https://github.com/knative/serving/issues/3049.
I also cleaned up some interface and linter errors, but this
code definitely needs much more love and care.
But this is outside of the scope.

* back to poor go code

* also kafka

sbezverk committed:
Clean up unit tests to use actual domain name (#797)

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

mattmoor committed:
Manually fixup prettier.io bugs. (#793)

nachocano committed:
Adding corev1.Events to GCP PubSub channel for easier debugging (#795)

* Emitting corev1.Events from gcppubsub channel.

* Fixing UTs and adding verbs to controller

* Updating dispatcher's event names.

nachocano committed:
Merge remote-tracking branch 'upstream/master' into broker-new

nachocano committed:
Changes to compile

nachocano committed:
moving filter

nachocano committed:
Moving cmds to broker

nachocano committed:
updating controller

nachocano committed:
Moving provider and reconciler to reconciler folder, and merging them into one

Harwayne committed:
Merge branch 'master' into broker-new-model

grantr committed:
Switch to new test image names (#800)

PR #725 removed paths from test image names, but the e2e tests were
still generating image URLs using the previous longer paths.

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into broker-new

nachocano committed:
Adding verbs to dispatcher

nachocano committed:
Adding create/update/patch events to gcp-pubsub dispatcher. (#801)

nachocano committed:
Moving back mains to cmd

nachocano committed:
Updating config

nachocano committed:
Moving resources to reconcilers

nachocano committed:
Moving broker folder back where it belongs

Harwayne committed:
Merge pull request #5 from nachocano/broker-new

Refactoring to new folder structure

nbarthwal committed:
Updating test-infra dependency (#802)

nachocano committed:
Merge branch 'broker-new-model' into broker-new

nachocano committed:
Revert "Merge branch 'broker-new-model' into broker-new"

This reverts commit d567412b062233463335c7feea94be795f7f2e15, reversing
changes made to bd6a8fa9c62460ee696854fb44a898e72e249ae2.

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into broker-new

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into filter-exp

nachocano committed:
Updating headers and headerExpressions to attributes and attributeExpressions

chaodaiG committed:
update deps:test-infra for fixing release.yaml not found error (#805)

continous and presubmit tests are failing due to release/yaml not found error, this is fixed in test-infra repo, update it here

Harwayne committed:
Merge branch 'master' into broker-clean-up

Harwayne committed:
Opt knative-eventing into Istio injection (for those whose Istio is defaulted to opt-out). (#806)

Harwayne committed:
Reconcilers notice when mroe things change.

Harwayne committed:
PR comment.

Harwayne committed:
Remove redundant watch.

Harwayne committed:
Merge pull request #6 from Harwayne/broker-clean-up

Broker clean up

Harwayne committed:
Unit test scaffold.

Harwayne committed:
Merge pull request #7 from Harwayne/broker-unit-test-scaffold

Broker unit test scaffold. They test almost nothing, but they pass, so gives a better starting point.

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into broker-new

Harwayne committed:
Tests for the namespace reconciler.

Harwayne committed:
Merge pull request #8 from Harwayne/ns-test

Tests for the namespace reconciler.

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into broker-new

Harwayne committed:
Broker controller unit tests.

nachocano committed:
Exact matching for filters without using k8s selectors-based syntax.
The matching can only be done on source and type for now.

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into filter-exp

nachocano committed:
Removing t3 as we don't have set expressions

mattmoor-sockpuppet committed:
golang format tools (#803)

Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -type f -name '*.go' -print))`
  `goimports -w $(find -name '*.go' | grep -v vendor)`

nachocano committed:
Updates after code review

sbezverk committed:
Adding header to the message being published (#794)

* Adding header to the message being published

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Adding header to the message being published

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Addressing comment

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Addressing comment

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Adding unit test

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* fixing unit test

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

nachocano committed:
Attempt to reconcile broker in trigger controller

Harwayne committed:
Merge pull request #9 from Harwayne/broker-test

Broker unit tests.

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into broker-new

Commit:
Update README.md (#810)

Harwayne committed:
Merge pull request #4 from nachocano/filter-exp

Trigger filters on event source and type

nachocano committed:
Deleting and re-creating subscription object as the backing channel
spec is immutable.

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into broker-new

nachocano committed:
Adding comment

Commit:
User can specify the namespace in yaml for knative-eventing (#711)

* User can specify the namespace in yaml for knative-eventing

* change ns to knative-testing

* update the test ns in reconcile_test.go

* change the test case for the new namespaces

* add clock.go in system

* Use knatve/pkg

Harwayne committed:
Rename Trigger.Spec.Filter.ExactMatch to Trigger.Spec.Filter.SourceAndType. Add unit tests for the filter binary.

Harwayne committed:
Merge pull request #11 from Harwayne/filter-test

Rename Trigger.Spec.Filter.ExactMatch to Trigger.Spec.Filter.SourceAndType. Add unit tests for the filter binary.

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into broker-new

nachocano committed:
Adding event messages

nachocano committed:
Using the broker's namespacedNamed as key to the triggers map.
With this we allow to reconcile only the triggers that belong to the
particular broker that changed

dprotaso committed:
bump knative/pkg - we drop webhook logic that increments spec.generation (#813)

Harwayne committed:
Merge pull request #10 from nachocano/broker-new

Reconcile Triggers on Broker changes.

mattmoor-sockpuppet committed:
golang format tools (#812)

Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -type f -name '*.go' -print))`
  `goimports -w $(find -name '*.go' | grep -v vendor)`

nachocano committed:
Adding some tests to trigger

nachocano committed:
More UTs

nachocano committed:
More UTs

nachocano committed:
More UTs

Harwayne committed:
Namespace reconciler automatically creates the Broker Filter's ServiceAccount and RBAC.

Sadly this doesn't work well because we have such an old version of controller-runtime that the Filter ends up trying to watch _all_ Triggers, not just those in its namespace. And it only gets permission for the Triggers in its own namespace.

Harwayne committed:
Merge pull request #12 from nachocano/broker-new-trigger-ut

Trigger controller UTs

Harwayne committed:
Remove no longer needed label.

nachocano committed:
Broker and trigger types UTs

grantr committed:
WIP early E2E test

Harwayne committed:
Switch the default channel provisioner from 'in-memory-channel' to 'in-memory'. (#790)

matzew committed:
:scream: for now, we skip some rules from mdl (#758)

sbezverk committed:
Restart dispatcher on connection loss (#796)

* Restart dispatcher on connection loss

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Addressing comments

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Refactor dispatcher for a dynamic reconnect

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Removing mutes

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Fixing unit test

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Fixing unit tests part 2

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Fixing unit tests part 3 racing issue

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Removing redundant mutex

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Address some issue in the logic

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Fixing logic a bit more

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Addressing comments

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Addressing more comments

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* fix racing in unit test

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Fixing a race

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Addressing comments

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Addressing racing conditions

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Addressing comments

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Addressing more comments

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

* Addressing more comments part3

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

nachocano committed:
Changes after code review. Adding trigger defaults and validation tests

nachocano committed:
Cleaner trigger validation

nachocano committed:
Adding dummy tests for broker validation...
Should be implemented

Harwayne committed:
Merge pull request #14 from nachocano/broker-new-trigger-ut

Broker and trigger types UTs

nachocano committed:
Merge remote-tracking branch 'grantr/broker-new-model-e2e' into broker-e2e

nachocano committed:
Compiling and moving things around

nbarthwal committed:
update dep for github.com/knative/test-infra along with fixes (#820)

nachocano committed:
Updating test

Commit:
Replace DOCKER_REPO_OVERRIDE by KO_DOCKER_REPO (#819)

nachocano committed:
More updates

nachocano committed:
Waiting for potentially multiple contents.
Removing check for corev1.Service ready.
Removing grant's great design. Just making it simpler for now.

sbezverk committed:
Switch to a non blocking send (#821)

Signed-off-by: Serguei Bezverkhi <sbezverk@cisco.com>

nachocano committed:
Compiling

nachocano committed:
Fixing compilation

evankanderson committed:
Fix env vars from #711 (#824)

nachocano committed:
Fixing compilation errors. Adding AnnotateNamespace function.

nachocano committed:
Adding ns

nachocano committed:
Adding logs. Changing to lowercase any otherwise the pod name is invalid

nachocano committed:
Removing namespace when creating trigger subscriber spec.
Adding wait time constant for default broker creation.

nachocano committed:
Checking if all triggers are ready

nachocano committed:
Updated logs

nachocano committed:
Working

nachocano committed:
Adding logs... Still not receiving the events.

nachocano committed:
More logs

nachocano committed:
Adding build constraint

nachocano committed:
Removing unnecessary stuff

nachocano committed:
Removing ugly structs

nachocano committed:
More logs

Harwayne committed:
Merge pull request #13 from Harwayne/ns-rbac

Namespace reconciler automatically creates the Broker Filter's ServiceAccount and RBAC.

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into broker-e2e

nachocano committed:
Removing quotes

nachocano committed:
More logs

Harwayne committed:
Add Harwayne as an approver. (#782)

nachocano committed:
Adding delay

nachocano committed:
Listing triggers in receiver when we create it, so that we don't miss
any message because the client couldn't find the existing trigger.
This is a problem in the in-memory-channel as it doesn't do retries.
Maybe the right solution is to add that functionality there.

nachocano committed:
Adding delay to sender pod

nachocano committed:
Removing withDelay method and just sleep for a while

nachocano committed:
Improve log...

mattmoor-sockpuppet committed:
golang format tools (#832)

Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -type f -name '*.go' -print))`
  `goimports -w $(find -name '*.go' | grep -v vendor)`

Harwayne committed:
ConfigMapWatcher.Start() blocks, rather than eagerly returning nil (#828)

* ConfigMapWatcher.Start() blocks, rather than eagerly returning nil.

* Move logging message closer to what it describes.

* Precreate the in-memory configMap.

* Print the time in a human readable format.

* hack/update-deps.sh

* PR comments.

nachocano committed:
Updates after code review.

nachocano committed:
Adding some more logs and trailing dots.

mattmoor-sockpuppet committed:
golang format tools (#838)

Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -type f -name '*.go' -print))`
  `goimports -w $(find -name '*.go' | grep -v vendor)`

Harwayne committed:
Bump Controller Runtime to 0.1.9 (#833)

* ConfigMapWatcher.Start() blocks, rather than eagerly returning nil.

* Move logging message closer to what it describes.

* Precreate the in-memory configMap.

* Print the time in a human readable format.

* hack/update-deps.sh

* PR comments.

* K8s 1.12.5 and Controller Runtime 0.1.9.

* hack/update-codegen.sh

* Update function call with the newly required parameter.

Harwayne committed:
Add labels to the stateful set. (#829)

Harwayne committed:
Merge branch 'master' into broker-controller-runtime

Harwayne committed:
Switch import order.

Harwayne committed:
Merge pull request #16 from Harwayne/broker-controller-runtime

Merge upstream/master.

Commit:
remove inlined and scothis from reviewers and approvers (#839)

nachocano committed:
Updating comments.

nachocano committed:
Updating comments.

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into broker-gen-pass

Harwayne committed:
Replace the bad errgroup usage with the runnableServer.

Harwayne committed:
Merge pull request #18 from Harwayne/codegen

Replace the bad errgroup usage with the runnableServer.

mattmoor-sockpuppet committed:
golang format tools (#822)

Produced via:
  `gofmt -s -w $(find -path './vendor' -prune -o -type f -name '*.go' -print))`
  `goimports -w $(find -name '*.go' | grep -v vendor)`

Harwayne committed:
Namespace scoped the Broker Filter's client.

Harwayne committed:
Fix unit tests.

Harwayne committed:
Fix yaml

Harwayne committed:
Merge pull request #19 from Harwayne/namespace-rolebinding

Namespace rolebinding, unit test fixes, and YAML fix.

Harwayne committed:
Remove the fake ListOptions that were used exclusively for limitations in the fake client. #833 updated the fake client to remove the need for these options. (#834)

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into broker-e2e

nachocano committed:
Merge branch 'broker-e2e' of github.com:nachocano/eventing into broker-e2e

nachocano committed:
Setting source to source not type.
Updating comment.

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into broker-gen-pass

Harwayne committed:
Merge pull request #15 from nachocano/broker-e2e

Broker-Trigger e2e test

Harwayne committed:
Merge pull request #17 from nachocano/broker-gen-pass

Broker-Trigger general pass

Harwayne committed:
Simplify Channel Watch (#835)

* ConfigMapWatcher.Start() blocks, rather than eagerly returning nil.

* Move logging message closer to what it describes.

* Precreate the in-memory configMap.

* Print the time in a human readable format.

* hack/update-deps.sh

* PR comments.

* K8s 1.12.5 and Controller Runtime 0.1.9.

* hack/update-codegen.sh

* Update function call with the newly required parameter.

* Switch to a simpler Channel Watch.

Harwayne committed:
Switch from annotating the namespace to labeling it, to match Istio.

Harwayne committed:
Merge pull request #20 from Harwayne/ns-eventing-label

Switch from annotating the namespace to labeling it, to match Istio.

adrcunha committed:
Update test-infra (#840)

Fixes [knative/test-infra#513](https://github.com/knative/test-infra/issues/513).

Major changes:
* more features in E2E tests
* properly fail releases when `ko` fails

Harwayne committed:
General clean up.

Harwayne committed:
Merge pull request #21 from Harwayne/broker-cleanup

General clean up.

Harwayne committed:
Merge branch 'master' into broker-merge-master

Harwayne committed:
Merge pull request #22 from Harwayne/broker-merge-master

Merge branch 'master' into broker-new-model

nak3 committed:
Fix invalid log message format (#845)

Currently metrics starting log is not correctly formatted as:

```
$ kubectl  -n knative-eventing logs eventing-controller-756d56fc7-g64cd
  ...
{"level":"info" ..snip.. "caller":"controller/main.go:140","msg":"Starting metrics listener at %s:9090","knative.dev/controller":"controller"}
```

This patch makes a tiny change to fix it, which replaces `logger.Info`
with `logger.Infof`.

nak3 committed:
Inform controller's metrics port is 9090 (#846)

Although event controller is opened 9090 port for metrics, there are
no information except for start logs.

This patch clarify the port in Deployment to be easily noticed by
users.

nak3 committed:
Update invalid docs link for telemetry (#844)

* Update invalid telemetry link in DEVELOPMENT.md

* Fix ordered list item by replacing all one

nak3 committed:
Fix config map name for kafka channel dispatcher (#850)

nak3 committed:
Fix flag parse error by calling flag.Parse (#851)

Currently kafka-channel-dispatcher outputs `ERROR: logging before
flag.Parse` messages.

This patch changes to calls flag.Parse() when program started to fix
the issue.

jcrossley3 committed:
Remove namespace from cluster-scoped resources (#853)

Commit:
fix lint problems (#857)

ultrasaurus committed:
add instructions on how to view roadmap (#817)

people need to join knative-users to view the roadmap doc

matzew committed:
Adding myself as a reviewer (#858)

Harwayne committed:
Initial docs for the Broker.

Harwayne committed:
Fill some of the usage section.

nak3 committed:
Use DeletionTimestamp field in subscription struct (#855)

* Use DeletionTimestamp field in subscription

This patch refactors by stop using meta.Accessor as subscription has
DeletionTimestamp field.

* Avoid declaring error variable in subscription reconcile

nak3 committed:
Use DeletionTimestamp field in channel struct (#856)

* Use DeletionTimestamp field in channel struct

This patch refactors by stop using meta.Accessor as channel struct has
DeletionTimestamp field.

* Avoid declaring error variable

mattmoor-sockpuppet committed:
Format markdown (#861)

Produced via: `prettier --write --prose-wrap=always $(find -name '*.md' | grep -v vendor | grep -v .github)`

akashrv committed:
Defaultcontroller (#843)

* Creating a default channel controller that will watch every channel
and update its status in case the channel is not being watched by any controller
This could happen if the end user creates a channel but doesn't install the
provisioner.
Issue#779: https://github.com/knative/eventing/issues/779

* Removed commetned lines

* Fixed some typos

* Changes based on comments on code review (pull request)

* Moved the code to set ChannelConditionProvisionerInstalled=True to Initialize()
function from MarkProvisioned()

* Changed all %v to %s when logging strings

* Updating some formatting issues called out in PR

Commit:
Distinguish config-logging (#842)

* Distinguish config-logging

* use env value get the configmap name

dubee committed:
Fix typos in docs (#864)

Harwayne committed:
Allow {Add,Remove}Finalizers to work on all Objects, not just Channels. (#865)

Harwayne committed:
Add instructions for installing the Broker without using Namespace annotation.

Harwayne committed:
Make ResolveSubscriberSpec a public method, so that it can be called by other Controllers that need to do the same resolution logic. (#866)

Harwayne committed:
Merge branch 'master' into broker-merge

Harwayne committed:
Merge pull request #29 from Harwayne/broker-merge

Merge branch 'master' into broker-new-model

Harwayne committed:
Merge pull request #28 from Harwayne/broker-docs

Broker and Trigger Docs

Harwayne committed:
Create example_{brokers,triggers}.yaml to document how they can be used.

Harwayne committed:
Merge pull request #30 from Harwayne/broker-cleanup

Create example_{brokers,triggers}.yaml to document how they can be used.

Harwayne committed:
Fix MD linter issues.

Harwayne committed:
Merge pull request #31 from Harwayne/md-fix

Fix MD linter issues.

Harwayne committed:
Fix MD linter issues.

Harwayne committed:
Merge pull request #32 from Harwayne/md-fix

Fix MD linter issues.

Harwayne committed:
Fix MD linter issues.

Harwayne committed:
Merge pull request #33 from Harwayne/md-fix

Fix MD linter issues.

Harwayne committed:
Minor clean up.

Harwayne committed:
Merge pull request #34 from Harwayne/broker-cleanup

Minor clean up.

matzew committed:
Add replica config (#869)

adrcunha committed:
Get the latest updates for automatic dot releases. (#874)

Harwayne committed:
Move the logging util out of contrib, in preparation for use by other… (#875)

* Move the logging util out of contrib, in preparation for use by other Eventing controllers using it.

* Move to pkg/logging.

Harwayne committed:
Subscription Controller uses zap.Logger (#876)

* Move the logging util out of contrib, in preparation for use by other Eventing controllers using it.

* Thread through context.

* Swap glog for zap.Logger in the Subscription reconciler.

* Pass in the logger form main().

Harwayne committed:
Increase unit test coverage of the Channel reconciler. (#878)

Harwayne committed:
Channel Controller uses zap.Logger (#877)

* Move the logging util out of contrib, in preparation for use by other Eventing controllers using it.

* Thread through context.

* Swap glog for zap.Logger in the Subscription reconciler.

* Pass in the logger form main().

* Channel controller uses zap.Logger.

* hack/update-deps.sh

Harwayne committed:
Merge branch 'master' into broker-merge

Harwayne committed:
Clean up some spots the merge didn't catch.

nak3 committed:
Do not call Fatal log during reconciling of kafka channel controller (#871)

* Do not call Fatal log during reconciling of kafka channel controller

Currently kafka channel controller keeps restarting if fails to build
admin client. Controller should try to reconcile without restart.

Having said that, users would like to notice the error easily. Hence
this patch introduces event logs for reconciler failures as well.

* Change info log for repeated success reconcile log with debug log

* Add warning event to unit test

nak3 committed:
Remove unnecessary printing format (#884)

This patch makes a tiny change by removing unnecessary printing
format `%s`.

Harwayne committed:
Merge pull request #35 from Harwayne/broker-merge

Merge branch 'master' into broker-new-model

Harwayne committed:
Update knative/pkg (#887)

* Update knative/pkg to 684620ff81ae137be5cda634a03831b373adc4eb (current HEAD).

* Remove the heart.

nak3 committed:
Return exit code kafka controller to avoid log lost (#889)

Since Kafka controller exits suddenly, controller does not call
`logger.Sync()` and lost log message sometimes.

This patch returns exit code to call `logger.Sync()` for sure.

Harwayne committed:
Merge branch 'master' into broker-merge-2

Harwayne committed:
Merge branch 'master' into broker-new-model

Merge branch 'master' into broker-new-model

Harwayne committed:
Fix the bad merge by replacing logger.BaseLogger with logger.FormatLogger.

Harwayne committed:
Merge pull request #37 from Harwayne/broker-merge-2

 Fix the bad merge by replacing logger.BaseLogger with logger.FormatLogger.

Harwayne committed:
Add extra columns when using kubectl get.

Harwayne committed:
MarkBrokerDoesNotExist

Harwayne committed:
Rename extra columns.

Harwayne committed:
Merge pull request #38 from Harwayne/broker-pr-comments

Add extra columns when using kubectl get.

yuzisun committed:
support partition consumer mode for kafka channel (#879)

* support partition consumer for kafka channel

* consolidate GetProvisionerConfig function

* add consumer mode config test

* address review comments

Harwayne committed:
Move ResolveSubscriberSpec() (#883)

* Move the logging util out of contrib, in preparation for use by other Eventing controllers using it.

* Thread through context.

* Swap glog for zap.Logger in the Subscription reconciler.

* Pass in the logger form main().

* Move ResolveSubscriberSpec out of Subscription Controller, as it will be used by multiple Controllers.

* Fix bad merge of gcp-pubsub.

yuzisun committed:
do not unsubscribe every time channel fanout config is updated for existing consumers (#880)

Harwayne committed:
Replace the Trigger reconciler's in-memory map with a simple list, utilizing the fact that Controller Runtime already has the results of the List cached.

Harwayne committed:
Merge pull request #39 from Harwayne/trigger-list

Replace the Trigger reconciler's in-memory map with a simple list

nachocano committed:
Accept v0.1 and v0.2 cloud events. Adding UTs.
Updating initClient as well, removing unnecessary paging.

Harwayne committed:
Create a mock dynamic client. (#902)

Harwayne committed:
Merge branch 'master' into broker-fix-merge

Harwayne committed:
Change to resolve.SubscriberSpec().

nachocano committed:
Merge remote-tracking branch 'broker/broker-new-model' into cloud_events

Harwayne committed:
Remove restClient as it wasn't actually used.

Harwayne committed:
Merge pull request #42 from Harwayne/remove-rest-config

Remove restClient as it wasn't actually used.

Harwayne committed:
Only reconcile the Namespace if the specific resource we care about changes.

Harwayne committed:
Merge pull request #43 from Harwayne/namespace-resources

Only reconcile the Namespace if the specific resource we care about changes

nachocano committed:
lowercase

Harwayne committed:
Merge pull request #40 from nachocano/cloud_events

Accept v0.1 and v0.2 cloud events. Adding UTs.

Harwayne committed:
Mark the Broker's Ingress and Filter status condidionts failed when they have failed.

Harwayne committed:
Merge pull request #45 from Harwayne/broker-mark-failed

Mark the Broker's Ingress and Filter status condidionts failed when they have failed.

Harwayne committed:
Do not DeepCopy() in Reconcile(), as controller-runtime already did it for us.

Harwayne committed:
Merge pull request #46 from Harwayne/broker-no-deepcopy

Do not DeepCopy() in Reconcile(), as controller-runtime already did it for us
